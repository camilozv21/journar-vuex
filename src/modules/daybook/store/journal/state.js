
export default () => ({
  isLoading: true,
  entries: [
    {
      id: '1',
      date: new Date().toDateString(),
      text: 'aLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
      picture: null,
    },
    {
      id: '2',
      date: new Date().toDateString(),
      text: 'bLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
      picture: null,
    },
    {
      id: '3',
      date: new Date().toDateString(),
      text: 'cLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
      picture: null,
    },
  ]
})