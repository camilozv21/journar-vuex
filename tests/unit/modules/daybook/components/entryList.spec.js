import { shallowMount } from '@vue/test-utils'
import { createStore } from 'vuex'

import journal from '@/modules/daybook/store/journal'
import EntryList from '@/modules/daybook/components/EntryList.vue'

import { journalState } from '../../../mock-data/test-journal-state'
import { test, suite } from "vitest";

const createVuexStore = ( initialState ) => 
createStore({
    modules: {
        journal: {
            ...journal,
            state: { ...initialState }
        }
    }
})

suite('Pruebas en el EntryList', () => {
    
    const store = createVuexStore( journalState )
    const mockRouter = {
        push: (...args) => { mockRouter.calls.push(args); } // Crear un espía manualmente
    }
    mockRouter.calls = []; // Inicializar el array de llamadas
    
    let wrapper

    const setup = () => {
        mockRouter.calls = []; // Limpiar los registros de la función espía
        wrapper = shallowMount( EntryList, {
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [ store ]
            }
        })
    }

    test('debe de llamar el getEntriesByTerm sin termino y mostrar 2 entradas', ({ expect }) => {
        setup();
        expect( wrapper.findAll('entry-stub').length ).toBe(2)
        expect( wrapper.html() ).toMatchSnapshot()

    })

    test('debe de llamar el getEntriesByTerm y filtrar las entradas', async({ expect }) => {
        setup();
        const input = wrapper.find('input')
        await input.setValue('segunda')

        expect( wrapper.findAll('entry-stub').length ).toBe(1)

    })
    
    test('el boton de nuevo debe de redireccionar a /new', ({ expect }) => {
        setup();
        wrapper.find('button').trigger('click')

        expect( mockRouter.calls[0] ).toEqual([{ name: 'entry', params: { id: 'new' } }])

    })
})