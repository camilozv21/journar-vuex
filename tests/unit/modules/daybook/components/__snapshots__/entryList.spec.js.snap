// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en el EntryList > debe de llamar el getEntriesByTerm sin termino y mostrar 2 entradas 1`] = `
"<div data-v-d2f66aca="" class="entry-list-container">
  <div data-v-d2f66aca="" class="px-2 pt-2"><input data-v-d2f66aca="" type="text" class="form-control" placeholder="Buscar entrada"></div>
  <div data-v-d2f66aca="" class="mt-2 d-flex flex-column"><button data-v-d2f66aca="" class="btn btn-primary mx-3"><i data-v-d2f66aca="" class="fa fa-plus-circle"></i> Nueva entrada </button></div>
  <div data-v-d2f66aca="" class="entry-scrollarea">
    <entry-stub data-v-d2f66aca="" entry="[object Object]"></entry-stub>
    <entry-stub data-v-d2f66aca="" entry="[object Object]"></entry-stub>
  </div>
</div>"
`;

exports[`Pruebas en el EntryList debe de llamar el getEntriesByTerm sin termino y mostrar 2 entradas 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar entrada"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva entrada </button></div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
